### PR TITLE
[systeminfo] Fix itests after core change

### PR DESCRIPTION
**Pull request by lolodomo**
Related to openhab/openhab-core#3600

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by lolodomo** [comment]
I am not sure if I have or not to call `registerService`.

**Comment by lolodomo** [comment]
Unfortunately, I can't build locally due to errors of this kind:
```
[ERROR] Tests not run because project has errors
[ERROR] Error   : Download java.io.IOException: Invalid content checksum 169E2D450E50B737F54582B0D89701022A8854ED550DBBB37C69EC81278999AF for file:/C:/Users/Laurent/.m2/repository/org/openhab/core/bundles/org.openhab.core.config.discovery/4.0.0-SNAPSHOT/org.openhab.core.config.discovery-4.0.0-SNAPSHOT.jar; expected BDF606DB0CBEBAF6FDB9FAF64BC279D59421981187A05ED4991C0FB83C2F5CE9 null
[ERROR] Error   : Download java.io.IOException: Invalid content checksum B8E2609C51D9EEB710409A80614507003E528AE8DD81FBA29D3E26F00DF96554 for file:/C:/Users/Laurent/.m2/repository/org/openhab/core/bundles/org.openhab.core/4.0.0-SNAPSHOT/org.openhab.core-4.0.0-SNAPSHOT.jar; expected EC95BE1CB0D9051C0A3977AB41A542D037E9A02E8387231A6AF0BA52FB3466CF null
...
```

**Comment by mherwege** [comment]
> Unfortunately, I can't build locally due to errors of this kind:
> 
> ```
> [ERROR] Tests not run because project has errors
> [ERROR] Error   : Download java.io.IOException: Invalid content checksum 169E2D450E50B737F54582B0D89701022A8854ED550DBBB37C69EC81278999AF for file:/C:/Users/Laurent/.m2/repository/org/openhab/core/bundles/org.openhab.core.config.discovery/4.0.0-SNAPSHOT/org.openhab.core.config.discovery-4.0.0-SNAPSHOT.jar; expected BDF606DB0CBEBAF6FDB9FAF64BC279D59421981187A05ED4991C0FB83C2F5CE9 null
> [ERROR] Error   : Download java.io.IOException: Invalid content checksum B8E2609C51D9EEB710409A80614507003E528AE8DD81FBA29D3E26F00DF96554 for file:/C:/Users/Laurent/.m2/repository/org/openhab/core/bundles/org.openhab.core/4.0.0-SNAPSHOT/org.openhab.core-4.0.0-SNAPSHOT.jar; expected EC95BE1CB0D9051C0A3977AB41A542D037E9A02E8387231A6AF0BA52FB3466CF null
> ...
> ```
> 
> even after a local rebuild of my core repo.

I got over these errors using `mvn install -U -DwithResolver -pl :org.openhab.binding.systeminfo.tests`, but then I run into:
```
[INFO] Resolving D:\openhab4-dev\git\openhab-addons\itests\org.openhab.binding.systeminfo.tests\itest.bndrun:
[ERROR] Resolution failed. Capabilities satisfying the following requirements could not be found:
    [<<INITIAL>>]
      ? osgi.identity: (osgi.identity=org.apache.servicemix.specs.activation-api-1.2.1)
    [org.openhab.binding.systeminfo.tests version=4.0.0.202305110950 type=osgi.fragment]
      ? osgi.wiring.package: (osgi.wiring.package=org.openhab.core.common.registry)
```

**Comment by mherwege** [comment]
> I am not sure if I have or not to call `registerService`.

I don't know, but If you have to, the better place is probably in the setUp method, so it only gets called once. I think it should also be wrapped in a waitForAssert. I have seen issues with integration tests failing because the service was not up yet at the moment fo acquiring the reference.

File: itests/org.openhab.binding.systeminfo.tests/src/main/java/org/openhab/binding/systeminfo/test/SysteminfoOSGiTest.java [hunk 1]
```diff
@@ -345,9 +347,15 @@ private void assertItemState(String acceptedItemType, String itemName, String pr
     }
 
     private void intializeItem(ChannelUID channelUID, String itemName, String acceptedItemType) {
+        UnitProvider unitProvider = getService(UnitProvider.class);
```
Comment (mherwege) [inline itests/org.openhab.binding.systeminfo.tests/src/main/java/org/openhab/binding/systeminfo/test/SysteminfoOSGiTest.java:350]: Suggest to make unitProvider a field, similar to itemRegistry and thingRegistry (line 132-133) and move this block to the setUp method (line 190).

**Comment by mherwege** [comment]
I pushed a PR to your repository with my adjustments. Feel free to evaluate and (not) include.

**Comment by lolodomo** [comment]
> I pushed a PR to your repository with my adjustments. Feel free to evaluate and (not) include.

Your pushed PR is invalid.
I propose to close this PR and then let you propose your own PR that I can then review.

**Pull request closed without merging**